lib/editor/components/HourMinuteInput.js

```diff
@@ -8,52 +8,49 @@ export default class HourMinuteInput extends Component {
     style: PropTypes.object
   }
 
-  state = {
-    seconds: typeof this.props.seconds === 'undefined' ? '' : this.props.seconds
-  }
+  state = {}
 
   _onChange = (evt) => {
     const {onChange} = this.props
     const {value} = evt.target
     const seconds = this.convertStringToSeconds(value)
-    if (seconds === this.state.seconds) {
-      this.setState({string: value})
-    } else {
-      this.setState({seconds, string: value})
-      onChange && onChange({seconds})
-    }
+    this.setState({string: value})
+    onChange && onChange(seconds)
+  }
+
+  componentWillReceiveProps (nextProps) {
+    this.setState({string: this.convertSecondsToString(nextProps.seconds)})
   }
 
   convertSecondsToString (seconds) {
     const hours = Math.floor(seconds / 60 / 60)
-    const minutes = Math.floor(seconds / 60)
+    const minutes = Math.floor(seconds / 60) % 60
     return seconds ? `${hours}:${minutes < 10 ? '0' + minutes : minutes}` : '00:00'
   }
 
   convertStringToSeconds (string) {
```

Comment:
This function doesn't rely on any instance variables, so please move it outside of this class.